src/Paramore.Brighter.MessagingGateway.AWSSQS/SqsMessageSender.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Amazon.SQS;
+using Amazon.SQS.Model;
+
+namespace Paramore.Brighter.MessagingGateway.AWSSQS;
+
+/// <summary>
+/// Class responsible for sending a message to a SQS
+/// </summary>
+public class SqsMessageSender
+{
+    private readonly string _queueUrl;
+    private readonly SnsSqsType _queueType;
+    private readonly AmazonSQSClient _client;
+
+    /// <summary>
+    /// Initialize the <see cref="SqsMessageSender"/>
+    /// </summary>
+    /// <param name="queueUrl">The queue ARN</param>
+    /// <param name="queueType">The queue type</param>
+    /// <param name="client">The SQS Client</param>
+    public SqsMessageSender(string queueUrl, SnsSqsType queueType, AmazonSQSClient client)
+    {
+        _queueUrl = queueUrl;
+        _queueType = queueType;
+        _client = client;
+    }
+
+    public async Task<string?> SendAsync(Message message, TimeSpan? delay, CancellationToken cancellationToken)
+    {
+        var request = new SendMessageRequest { QueueUrl = _queueUrl, MessageBody = message.Body.Value, };
+
+        if (delay != null)
+        {
+            request.DelaySeconds = (int)delay.Value.TotalSeconds;
+        }
+
+        if (_queueType == SnsSqsType.Fifo)
+        {
+            request.MessageGroupId = message.Header.PartitionKey;
+            if (message.Header.Bag.TryGetValue(HeaderNames.DeduplicationId, out var deduplicationId))
+            {
+                request.MessageDeduplicationId = (string)deduplicationId;
+            }
+        }
+
+        var messageAttributes = new Dictionary<string, MessageAttributeValue>
```

Comment:
Typically we factor this into a publisher, to keep the producer clean of all the header manipulation